src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -32,6 +32,7 @@ protected override async Task<SyntaxNode> GetSelectedNodeAsync(Document document
         {
             var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
 
+            // Consider MemberDeclaration (types, methods, events, delegates, ...) and VariableDeclarator (pure variables)
             var memberDecl = await refactoringHelperService.TryGetSelectedNodeAsync<MemberDeclarationSyntax>(document, span, cancellationToken).ConfigureAwait(false);
```

Comment:
so, not your fault (since you're working on code that wasn't yours).  However, the more precise explanation here is something more like:

> In general, we want to handle members that can be declared in a type (since those are what we would be pulling up).  These are generally MemberDeclarations (like methods/properties/etc.), so we just check directly for that.  However, in the case of fields, things are stored slightly differently.  A field declaration might be `int a = 0, b = 1;`.  The MemberDeclaration would represent the entirety of both those fields.  However a user might just want to select one of them, so we also look for the nearest VariableDeclarator (i.e. `a = 0`) that they might be selecting.  That way they can just pull up a single field from a set in a declaration.

The intent is not to transliterate what hte C# code says, but to give enogh context for a future reader (including yourself) to understand the *whys* of it.  i.e. "why was a VariableDeclarator important?  Why was MemberDeclaration not enough?".   Stating `and VariableDeclarator (pure variables)` doesn't explain that to me as a reader.

i hope that helps.

(and yes, this shouldn't be your responsibility.  but often we go by "the last one who touches it needs to fix it" :))

Reply:
> The intent is not to transliterate what hte C# code says, but to give enogh context for a future reader

Sure, I get that however since I didn't have all the relevant background I wanted to note at least what I could gather from quickly reading / working with the code. 

I actually didn't realize that the intend was to enable selecting sub-parts (tho it makes sense now thinking about it). Will update accordingly.